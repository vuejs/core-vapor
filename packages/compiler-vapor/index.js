'use strict'

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./dist/compiler-vapor.cjs.prod.js')
} else {
  module.exports = require('./dist/compiler-vapor.cjs.js')
}
