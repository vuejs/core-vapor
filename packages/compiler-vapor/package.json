{
  "name": "@vue/compiler-vapor",
  "version": "0.0.0-vapor",
  "description": "@vue/compiler-vapor",
  "main": "index.js",
  "module": "dist/compiler-vapor.esm-bundler.js",
  "types": "dist/compiler-vapor.d.ts",
  "unpkg": "dist/compiler-vapor.global.js",
  "jsdelivr": "dist/compiler-vapor.global.js",
  "files": [
    "index.js",
    "dist"
  ],
  "sideEffects": false,
  "buildOptions": {
    "name": "VueCompilerVapor",
    "compat": true,
    "formats": [
      "esm-bundler",
      "esm-browser",
      "cjs",
      "global"
    ]
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vuejs/core-vapor.git",
    "directory": "packages/compiler-vapor"
  },
  "keywords": [
    "vue"
  ],
  "author": "Evan You",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/vuejs/core-vapor/issues"
  },
  "homepage": "https://github.com/vuejs/core-vapor/tree/main/packages/compiler-vapor#readme",
  "dependencies": {
    "@vue/compiler-dom": "workspace:*",
    "@vue/shared": "workspace:*",
    "source-map-js": "^1.0.2"
  }
}
