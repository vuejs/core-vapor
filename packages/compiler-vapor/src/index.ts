export { parse } from '@vue/compiler-dom'
export { transform } from './transform'
export { generate } from './generate'
export { compile, type CompilerOptions } from './compile'
export * from './ir'
export * from './errors'
