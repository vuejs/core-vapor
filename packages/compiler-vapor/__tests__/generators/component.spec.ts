import { compile } from '@vue/compiler-vapor'

describe('generate component', () => {
  test('generate single root component (without props)', () => {
    const { code } = compile(`<Comp/>`)
    expect(code).toMatchSnapshot()
  })

  test('generate single root component (with props)', () => {
    const { code } = compile(`<Comp :foo="foo"/>`)
    expect(code).toMatchSnapshot()
  })

  test('generate multi root component', () => {
    const { code } = compile(`<Comp/>123`)
    expect(code).toMatchSnapshot()
  })

  test('should not generate withAttrs if component is not the root of the template', () => {
    const { code } = compile(`<div><Comp/></div>`)
    expect(code).toMatchSnapshot()
  })
})
