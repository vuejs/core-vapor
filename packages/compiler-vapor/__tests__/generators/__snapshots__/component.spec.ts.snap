// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generate component > generate multi root component 1`] = `
"import { resolveComponent as _resolveComponent, createComponent as _createComponent, template as _template } from 'vue/vapor';
const t0 = _template("123")

export function render(_ctx) {
  const n1 = t0()
  const n0 = _createComponent(_resolveComponent("Comp"))
  return [n0, n1]
}"
`;

exports[`generate component > generate single root component (with props) 1`] = `
"import { resolveComponent as _resolveComponent, createComponent as _createComponent } from 'vue/vapor';

export function render(_ctx) {
  const n0 = _createComponent(_resolveComponent("Comp"), [{
    foo: () => (foo)
  }], true)
  return n0
}"
`;

exports[`generate component > generate single root component (without props) 1`] = `
"import { resolveComponent as _resolveComponent, createComponent as _createComponent } from 'vue/vapor';

export function render(_ctx) {
  const n0 = _createComponent(_resolveComponent("Comp"), null, true)
  return n0
}"
`;

exports[`generate component > should not generate withAttrs if component is not the root of the template 1`] = `
"import { resolveComponent as _resolveComponent, createComponent as _createComponent, insert as _insert, template as _template } from 'vue/vapor';
const t0 = _template("<div></div>")

export function render(_ctx) {
  const n1 = t0()
  const n0 = _createComponent(_resolveComponent("Comp"))
  _insert(n0, n1)
  return n1
}"
`;
